src/Libraries/UnitsUI/NodeViewCustomizations.cs

```diff
@@ -0,0 +1,684 @@
+                }
+            };
+
+            if (precisionCB.SelectedItem == null)
+            {
+                precisionCB.SelectedIndex = RetrieveSelectedItemIndex(unitValueDropdownViewModel.SelectedPrecision.ToString(), precisionCB.Items);
+            }
+
+            dockPanelPrecision.Children.Add(precisionLabel);
+            dockPanelPrecision.Children.Add(precisionCB);
+            listView.Items.Add(dockPanelPrecision);
+            //obsCollection.Add(dockPanelPrecision);
+
+            var dockPanelFormat = new DockPanel();
+            dockPanelFormat.Background = null;
+            dockPanelFormat.FlowDirection = FlowDirection.LeftToRight;
+            var formatLabel = new Label
+            {
+                Content = "Format",
+                MinWidth = labelMinWidth
+            };
+
+            ComboBox formatCB = new ComboBox
+            {
+                DataContext = unitValueDropdownViewModel,
+                FlowDirection = FlowDirection.LeftToRight,
+                ItemsSource = unitValueDropdownViewModel.AllFormats,
+                MinWidth = comboMinMidth,
+                MaxWidth = comboMaxWidth,
+                HorizontalAlignment = HorizontalAlignment.Stretch,
+                ToolTip = unitValueDropdownViewModel.SelectedFormat
+            };
+
+            // bind this combo box to the selected item hash
+            var bindingValAllFormats = new System.Windows.Data.Binding(nameof(UnitValueOutputDropdownViewModel.AllFormats))
+            {
+                Source = unitValueDropdownViewModel
+            };
+            formatCB.SetBinding(ItemsControl.ItemsSourceProperty, bindingValAllFormats);
+
+            // bind the selected index to the model property SelectedIndex
+            var indexBindingSelectedFormat = new Binding(nameof(unitValueDropdownViewModel.SelectedFormat))
+            {
+                Mode = BindingMode.TwoWay,
+                Source = unitValueDropdownViewModel
+            };
+            formatCB.SetBinding(Selector.SelectedItemProperty, indexBindingSelectedFormat);
+            formatCB.SelectionChanged += delegate
+            {
+                if (formatCB.SelectedIndex != -1)
+                {
+                    RaisePropertyChanged(nameof(UnitValueOutputDropdownViewModel.SelectedFormat));
+                    formatCB.ToolTip = unitValueDropdownViewModel.SelectedFormat;
+                }
+            };
+
+            if (formatCB.SelectedItem == null)
+            {
+                formatCB.SelectedIndex = RetrieveSelectedItemIndex(unitValueDropdownViewModel.SelectedFormat.ToString(), formatCB.Items);
+            }
+
+            dockPanelFormat.Children.Add(formatLabel);
+            dockPanelFormat.Children.Add(formatCB);
+            listView.Items.Add(dockPanelFormat);
+
+            ex.Content = listView;
+            nodeView.inputGrid.Children.Add(grid);
+        }
+
+        int RetrieveSelectedItemIndex(string name, ItemCollection items)
```

Comment:
do we need this? could we not use `IndexOf`?

Reply:
Absolutely. Done.